pkg/apis/kubernikus/factory.go

```diff
@@ -53,8 +53,14 @@ func (klusterFactory) KlusterFor(spec models.KlusterSpec) (*v1.Kluster, error) {
 		},
 	}
 
-	if k.Spec.ClusterCIDR == "" {
-		k.Spec.ClusterCIDR = DEFAULT_CLUSTER_CIDR
+	//If the cluster CIDR was not set at all we default it,
+	//if was set to empty string explicitly we null it
```

Comment:
this is weird. i have no clue if this holds up in the templates
i think most / all clusters with empty CIDR have no nodes but i am unsure if we actually prevent it.

Reply:
I made an additional change that you can only set an empty clusterCIDR if `noCloud: true`, so its unlikely that this should happened by accident.